src/pages/api/index.ts

```diff
@@ -19,7 +20,43 @@ const endpoint: NextApiHandler = async (req, res) => {
 		apolloServerHandler = apolloServer.createHandler({
 			path: "/api",
 		});
-		console.log("Server handler started ✅");
+	}
+	if (req.session.user) {
+		/*
+		By default the session.user objects only contains the id
+		but usually we need more, especially the permission when deciding
+		whether or not the user is authorized to see/content.
+		For this reason, when a user is authorized we query the user object
+		with all its relations and replace the session.user object with it.
```

Comment:
This queries the user's entire relation tree every single time a query is sent to the API. I'd suggest we just find the necessary data for every resolver instead.

Reply:
Yes, lets to that